frontend/src/Editor/QueryManager/QueryManager.jsx

```diff
@@ -468,46 +458,87 @@ class QueryManagerComponent extends React.Component {
-                    >
-                      &nbsp; {this.props.t('editor.queryManager.advanced', 'Advanced')}
-                    </a>
-                  </li>
-                </ul>
-              </div>
+              <>
+                <span
+                  className={`${
+                    this.props.darkMode ? 'color-light-gray-c3c3c3' : 'color-light-slate-11'
+                  } cursor-pointer font-weight-400`}
+                  onClick={() => {
+                    if (mode === 'edit') {
+                      this.setState({
+                        selectedDataSource: null,
+                        selectedQuery: {},
+                        options: {},
+                        isSourceSelected: false,
+                        mode: 'create',
+                      });
+                    } else {
+                      this.setState({
+                        isSourceSelected: false,
+                        selectedDataSource: null,
+                        options: {},
+                      });
+                    }
+                  }}
+                >
+                  {mode === 'create' ? 'New Query' : 'Queries'}
+                </span>
+                <span className="breadcrum">
+                  <svg width="5.33" height="9.33" viewBox="0 0 8 14" fill="none" xmlns="http://www.w3.org/2000/svg">
+                    <path
+                      fillRule="evenodd"
+                      clipRule="evenodd"
+                      d="M0.292893 0.292893C0.683417 -0.0976311 1.31658 -0.0976311 1.70711 0.292893L7.70711 6.29289C8.09763 6.68342 8.09763 7.31658 7.70711 7.70711L1.70711 13.7071C1.31658 14.0976 0.683417 14.0976 0.292893 13.7071C-0.0976311 13.3166 -0.0976311 12.6834 0.292893 12.2929L5.58579 7L0.292893 1.70711C-0.0976311 1.31658 -0.0976311 0.683417 0.292893 0.292893Z"
+                      fill="#C1C8CD"
+                    />
+                  </svg>
+                </span>
+                <div className="query-name-breadcrum d-flex align-items-center">
+                  <span className="query-manager-header-query-name font-weight-400">
+                    {this.state.renameQuery ? (
+                      <input
+                        type="text"
+                        className={`query-name query-name-input-field border-indigo-09 bg-transparent  ${
+                          this.props.darkMode && 'text-white'
+                        }`}
+                        autoFocus
+                        defaultValue={queryName}
+                        onChange={({ target }) => (this.newNameForQuery.current = target.value)}
+                        onKeyUp={(event) => {
+                          event.persist();
+                          if (event.keyCode === 13) {
+                            this.executeQueryNameUpdation();
+                          } else {
+                            this.debounceUpdateQueryName();
+                          }
```

Comment:
Debounce is not need in this case. We can save the query name with `onBlur` event
```suggestion
                          } 
```